### DISCUSSION TITLE
how can i set a keymap which using neogit push

### DISCUSSION THREAD

**Discussion by rohi-zalcman** [NeogitOrg/neogit#1514] [Q&A]
Hello, I am trying to set a keymap for using Neogit to push. I tried the following:
{
    '<leader>gp',
    '<cmd> lua require("neogit").action("push", "to_pushremote", {})<cr>',
    desc = 'NeoGit push',
}
Could someone help me with this?

**Comment by CKolkey** [comment]
```lua
{
  'gp',
  require("neogit").action("push", "to_pushremote", {}),
  desc = 'NeoGit push',
}
```

The value is not a string, but just lua.

**Reply by rohi-zalcman** [reply]
sorry this is was the orignal:
      {
        '<leader>gp',
        '<cmd> lua require("neogit").action("push", "to_pushremote", {})<cr>',
        desc = 'NeoGit push',
      },
but its still not work

**Reply by CKolkey** [reply]
Replace 
```lua
' lua require("neogit").action("push", "to_pushremote", {})',
``` 
with 
```lua
require("neogit").action("push", "to_pushremote", {}),
```
 :)

**Reply by rohi-zalcman** [reply]
yep i tried but nothing happen

**Reply by CKolkey** [reply]
If you just enter `:lua require("neogit").action("push", "to_pushremote", {})()` as a vim command, you should see a notification appear that it's pushing. Is that right?

**Comment by rohi-zalcman** [comment]
Yep but it does  nothing
When I type it as git command
